### PR TITLE
Use io.BytesIO instead of io.StringIO in examples

### LINKED ISSUE #6201: examples svg_histogram.html failes with UnicodeEncodeError
[Your example](http://matplotlib.org/examples/user_interfaces/svg_histogram.html) fails with a UnicodeEncodeError

    python svg_histogram.py

results in:

    Traceback (most recent call last):
      File "svg_histogram.py", line 83, in <module>
        tree, xmlid = ET.XMLID(f.getvalue())
      File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1316, in XMLID
        parser.feed(text)
      File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1640, in feed
        self._parser.Parse(data, 0)
    UnicodeEncodeError: 'ascii' codec can't encode character u'\u2212' in position 6447: ordinal not in range(128)
    
Thanks!

- Matplotlib version '1.5.1', Python version 2.7.9 on both platforms Windows, & Linux
- pip install matplotlib

### PR DESCRIPTION
**Pull request by mdboom**
Fixes #6201.

### CODE REVIEW

File: examples/misc/svg_filter_line.py [hunk 1]
```diff
@@ -52,7 +52,7 @@
 ax.set_ylim(0., 1.)
 
 # save the figure as a string in the svg format.
-from io import StringIO
+from six import StringIO
```
Comment (WeatherGod) [inline examples/misc/svg_filter_line.py]: should we use the internal six?

Comment (tacaswell) [inline examples/misc/svg_filter_line.py]: Yes.

Comment (tacaswell) [inline examples/misc/svg_filter_line.py]: er, no this is in the examples, we should not be teaching anyone to use our internal six.

**Comment by QuLogic** [comment]
I still think it'd be better to switch to `BytesIO` in all cases instead of introducing `six` in the examples.

**Comment by tacaswell** [comment]
:+1: on less use of `six`

**Comment by mdboom** [comment]
I've updated this (and the title of this issue) to use `io.BytesIO` instead.

File: doc/faq/howto_faq.rst [hunk 2]
```diff
@@ -664,8 +664,9 @@ Here is an example using `Pillow <http://python-imaging.github.io/>`_.
 First, the figure is saved to a StringIO object which is then fed to
```
Comment (efiring) [inline doc/faq/howto_faq.rst]: Change here also.

File: doc/faq/howto_faq.rst [hunk 3]
```diff
@@ -664,8 +664,9 @@ Here is an example using `Pillow <http://python-imaging.github.io/>`_.
...
 Pillow for further processing::
 
-    import StringIO, Image
-    imgdata = StringIO.StringIO()
+    from io import BytesIO
+    import Image
```
Comment (QuLogic) [inline doc/faq/howto_faq.rst]: Also, while you're here, should this not be `from PIL import Image`?

File: examples/misc/svg_filter_line.py [hunk 4]
```diff
@@ -52,8 +52,8 @@
 ax.set_ylim(0., 1.)
 
 # save the figure as a string in the svg format.
```
Comment (QuLogic) [inline examples/misc/svg_filter_line.py]: Comment still says string.

**Comment by tacaswell** [comment]
backported to 1.5.1-doc as 40cfef1

**Pull request merged**